src/Bandits/BanditEvaluator.php

```diff
@@ -0,0 +1,268 @@
+<?php
+
+namespace Eppo\Bandits;
+
+use Eppo\DTO\Bandit\ActionCoefficients;
+use Eppo\DTO\Bandit\AttributeSet;
+use Eppo\DTO\Bandit\BanditEvaluation;
+use Eppo\DTO\Bandit\BanditModelData;
+use Eppo\DTO\Bandit\NumericAttributeCoefficient;
+use Eppo\Exception\BanditEvaluationException;
+use Eppo\Exception\InvalidArgumentException;
+use Eppo\Sharder;
+
+class BanditEvaluator implements IBanditEvaluator
+{
+    public function __construct(private readonly int $totalShards = 10000)
+    {
+    }
+
+
+    /**
+     * @param string $flagKey
+     * @param string $subjectKey
+     * @param AttributeSet $subject
+     * @param array<string, AttributeSet> $actionsWithContexts
+     * @param BanditModelData $banditModel
+     * @return BanditEvaluation
+     * @throws BanditEvaluationException
+     * @throws InvalidArgumentException
+     */
+    public function evaluateBandit(
+        string $flagKey,
+        string $subjectKey,
+        AttributeSet $subject,
+        array $actionsWithContexts,
+        BanditModelData $banditModel
+    ): BanditEvaluation {
+        if (empty($actionsWithContexts)) {
+            throw new InvalidArgumentException("No actions provided for bandit evaluation");
+        }
```

Comment:
We only want to do this if the bandit is still at play (e.g., this flag has a bandit).

This is to support the case of users removing bandits from the flag more in a friendly way, such as rolling out a winning variation. That way, if they just stop supplying actions but leave this in, it will still work as they expect.

You can see the node example here: https://github.com/Eppo-exp/js-client-sdk-common/blob/main/src/client/eppo-client.ts#L471

Reply:
This is actually handled up in the [`EppoClient`](https://github.com/Eppo-exp/php-sdk/pull/29/files#diff-40894f9232b859c74a1416e75c4eeeb48fd50940ed2dbb4c7f9e31878cdf1a17R484). My reasoning for throwing an exception here is that if we're invoking the BanditEvaluator, we've already matched an assignment to a bandit, pulled the Bandit model and should not have done all that work if the action list is empty.

Reply:
This will return but not log the default variation. It sounds like our latest plan is for us to wait to check for zero actions until a variation has been assigned?